src/qibo/models/variational.py

```diff
@@ -97,6 +97,142 @@ def _loss(params, circuit, hamiltonian):
         return result, parameters, extra
 
 
+class AAVQE(object):
+    """This class implements the Adiabatically Assisted Variational Quantum Eigensolvers
+    algorithm. See https://arxiv.org/abs/1806.02287.
+
+    Args:
+        circuit (:class:`qibo.abstractions.circuit.AbstractCircuit`): variaional ansatz.
+        easy_hamiltonian (:class:`qibo.hamiltonians.Hamiltonian`): initial Hamiltonian object.
+        problem_hamiltonian (:class:`qibo.hamiltonians.Hamiltonian`): problem Hamiltonian object.
+        s (callable): scheduling function of time that defines the scheduling of the
+            adiabatic evolution. It must verify boundary contitions: s(0) = 0 and s(1) = 1.
+        nsteps (float): number of steps of the adiabatic evolution.
+        t_max (float): total time of time evolution.
+
+    Example:
+    ::
+        import numpy as np
+        from qibo import gates, models, hamiltonians
+        # create circuit ansatz for two qubits
+        circuit = models.Circuit(2)
+        circuit.add(gates.RY(0, theta=0))
+        circuit.add(gates.RY(1, theta=0))
+        # define the easy and the problem Hamiltonian.
+        easy_hamiltonian=hamiltonians.X(2)
+        problem_hamiltonian=hamiltonians.XXZ(2)
+        # define a scheduling function with only one parameter 
+        # and boundary conditions s(0) = 0, s(1) = 1
+        s = lambda t: t
+        # create AAVQE model 
+        aavqe = models.AAVQE(circuit, easy_hamiltonian, problem_hamiltonian, 
+                             s, nsteps = 10, Tmax = 1)
+        # optimize using random initial variational parameters
+        np.random.seed(0)
+        initial_parameters = np.random.uniform(0, 2*np.pi, 2)
+        ground_energy, params = aavqe.minimize(initial_parameters)
+    """
+
+    def __init__(self, circuit, easy_hamiltonian, problem_hamiltonian, s, nsteps=10, t_max = 1):
+        from qibo.abstractions import hamiltonians
+
+        if not issubclass(type(easy_hamiltonian), hamiltonians.HAMILTONIAN_TYPES):
+            raise_error(TypeError, "The easy Hamiltonian should be a hamiltonians.Hamiltonian "
+                                   "object but is {}.".format(type(easy_hamiltonian)))
+        if type(problem_hamiltonian) != type(easy_hamiltonian):
+            raise_error(TypeError, "The problem Hamiltonian should be of the same type of the easy Hamiltonian"
+                                    " ({}), but it is {}.".format(type(easy_hamiltonian), type(problem_hamiltonian)))
+        if easy_hamiltonian.nqubits != problem_hamiltonian.nqubits:
+            raise_error(ValueError, "The easy Hamiltonian has {} qubits while problem Hamiltonian has {}."
+                                    "".format(easy_hamiltonian.nqubits, problem_hamiltonian.nqubits))
+
+        self.ATOL = 1e-7 # Tolerance for checking s(0) = 0 and s(T) = 1.
+        self.ATOL_TIME = 1e-7 # Tolerance for checking if time is greater than t_max.
```

Comment:
Please move both attributes to new arguments (with default value of `1e-7`) in the `__init__` function.